src/linter.ts

```diff
@@ -0,0 +1,110 @@
+import { languages, DiagnosticCollection } from 'vscode';
+import { Diagnostic, DiagnosticSeverity, Disposable, Range, TextDocument } from 'vscode';
+import * as context from './context';
+import * as wrapper from './wrappers';
+
+export class Linter implements Disposable {
+    private _timer: NodeJS.Timer;
+    constructor(private _context: context.Context,
+                private _wrapper: wrapper.VSCode,
+                private _diagnosticCollection: DiagnosticCollection) {
+    }
+
+    public start(): void {
+        clearTimeout(this._timer);
```

Comment:
I think I would prefer the "do not re-start timer if one is running" behavior.

With the behavior you implemented, the linter won't run until there is no activity for a second.

With the other method, we will have a lint once every one second:

~~~
if (this._timer === null) {
    this._timer = setTimeout(() => {
         this._timer = null;
         this.lintDocument();
    }, 1000);
}
~~~

What do you think ?

Reply:
My plan is to run the linter when the document is edited rather than on save (the reason it's not like that now is because `autoprojExec()` fails for some odd reason when `-` is passed as an argument). For such a use case, I think waiting for no activity is better because we ensure we are linting the most up-to-date document

Reply:
OK. Let's try your way and update accordingly if we don't like it.